sklearn/preprocessing/data.py

```diff
@@ -1689,15 +1697,15 @@ class OneHotEncoder(BaseEstimator, TransformerMixin):
 
     Parameters
     ----------
-    n_values : 'auto', int or array of ints
-        Number of values per feature.
-
-        - 'auto' : determine value range from training data.
-        - int : number of categorical values per feature.
-                Each feature value should be in ``range(n_values)``
-        - array : ``n_values[i]`` is the number of categorical values in
-                  ``X[:, i]``. Each feature value should be
-                  in ``range(n_values[i])``
+    values : 'auto', 'auto-strict', int, List[int], or List[List[objects]]
+        - 'auto' (default) : Determine set of values from training data.
+            If values are integers, then allowed values will be between
```

Comment:
we have a distinction between "allowed" and "transformed" values for "auto" with ints. Maybe specify "errors will only be raised in transform for values out of the input range, when the input is integer". Really, though, this is a weird API, which is why we had designed `handle_unknown="error-strict"`. Perhaps it's easier to change `handle_unknown="error-strict"` to become the default in a future version than `values="auto-strict"`. WDYT?